openshift/ansible-service-broker
# Add ASB operator deployment manifest assets.

#### Describe what this PR does and why we need it:
Move ASB operator manifests from my *[stuff](https://github.com/djwhatle/stuff/tree/master/olm-svcat-brokers-testing)* repo to here, and add additional resources to create the namespace and clusterrolebinding. This _is not_ what we want the final state of the `manifests` directory to be, but it is a step towards centralizing the location of these assets.

Still need to test if `oc create -f manifests` works to get ASB running on a 4.0 cluster with OLM. I already know that the CR can't be created immediately since the CRD won't be registered until some amount of time after the ASB Operator Subscription is created.

Note that this PR separates the previous `osb-operators` configmap into individual CatalogSources/ConfigMaps for both ASB and TSB operators.


#### To test:
 - From a fresh 4.0 cluster using tagged installer release v0.7.0 or newer, (or one completely cleaned of the old `osb-operators` ASB/TSB combined CatalogSource / automation-broker namespace), run `oc create -f ansible-service-broker/manifests`.
 - The first time you run the above command, you should get a few "failed to create resource" messages. This is expected. It takes some time for OLM to create and execute the InstallPlan that will install the `automationbroker` CRD onto the cluster.
 - Wait a few seconds and run the same command `oc create -f ansible-service-broker/manifests` again, this time you should see the remaining resources create successfully, and you should get an `automation-broker` pod running after some time.
